# Análisis Exhaustivo del Dataset de Alimentos

In [2]:
import pandas as pd

# Load the dataset
file_path = 'FOOD-DATA-GROUP1.csv'
data = pd.read_csv(file_path)

# Display basic information about the dataset
data_info = data.info()
data_description = data.describe()

# Extract a brief description of what five columns represent
columns_description = {
    'Caloric Value': 'Representa la cantidad de calorías en el alimento, medida en kilocalorías (kcal).',
    'Fat': 'Indica la cantidad total de grasa en el alimento, medida en gramos (g).',
    'Carbohydrates': 'Mide la cantidad de carbohidratos presentes en el alimento, en gramos (g).',
    'Protein': 'Representa la cantidad de proteínas en el alimento, en gramos (g).',
    'Sodium': 'Indica la cantidad de sodio en el alimento, medida en miligramos (mg).',
}

data_info, data_description, columns_description

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 551 entries, 0 to 550
Data columns (total 37 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0.1          551 non-null    int64  
 1   Unnamed: 0            551 non-null    int64  
 2   food                  551 non-null    object 
 3   Caloric Value         551 non-null    int64  
 4   Fat                   551 non-null    float64
 5   Saturated Fats        551 non-null    float64
 6   Monounsaturated Fats  551 non-null    float64
 7   Polyunsaturated Fats  551 non-null    float64
 8   Carbohydrates         551 non-null    float64
 9   Sugars                551 non-null    float64
 10  Protein               551 non-null    float64
 11  Dietary Fiber         551 non-null    float64
 12  Cholesterol           551 non-null    float64
 13  Sodium                551 non-null    float64
 14  Water                 551 non-null    float64
 15  Vitamin A             5

(None,
        Unnamed: 0.1  Unnamed: 0  Caloric Value         Fat  Saturated Fats  \
 count    551.000000  551.000000     551.000000  551.000000      551.000000   
 mean     275.000000  275.000000     237.359347   10.766933        3.722715   
 std      159.204271  159.204271     199.235598   12.544196        5.397021   
 min        0.000000    0.000000       3.000000    0.000000        0.000000   
 25%      137.500000  137.500000      94.500000    2.500000        0.500000   
 50%      275.000000  275.000000     186.000000    6.600000        1.800000   
 75%      412.500000  412.500000     337.000000   14.400000        4.800000   
 max      550.000000  550.000000    1578.000000   87.500000       43.500000   
 
        Monounsaturated Fats  Polyunsaturated Fats  Carbohydrates      Sugars  \
 count            551.000000            551.000000     551.000000  551.000000   
 mean               4.001564              2.170793      15.819036    2.708907   
 std                5.540608         

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

# Load the dataset
file_path = 'FOOD-DATA-GROUP1.csv'
data = pd.read_csv(file_path)

# Display basic information about the dataset
print("Información del Dataset:")
data_info = data.info()

# Display the first few rows of the dataset
print("\nPrimeras filas del Dataset:")
first_rows = data.head()

# Display descriptive statistics of the dataset
print("\nEstadísticas Descriptivas del Dataset:")
data_description = data.describe()

# Select only numerical columns for correlation matrix
numeric_data = data.select_dtypes(include=['float64', 'int64'])

# Calculate the correlation matrix
correlation_matrix = numeric_data.corr()

# Display the correlation matrix
print("\nMatriz de Correlación:")
correlation_matrix_display = correlation_matrix

# Visualize the correlation matrix using a heatmap
plt.figure(figsize=(16, 12))
heatmap = sns.heatmap(correlation_matrix, annot=True, cmap='coolwarm', fmt='.2f', linewidths=0.5)
plt.title('Mapa de Calor de la Matriz de Correlación')
plt.show()

# Select columns with significant correlations
correlation_threshold = 0.5
significant_correlations = correlation_matrix[abs(correlation_matrix) > correlation_threshold].stack().reset_index()
significant_correlations = significant_correlations[significant_correlations['level_0'] != significant_correlations['level_1']]
significant_correlations.columns = ['Variable 1', 'Variable 2', 'Correlación']

print("\nCorrelaciones Significativas (absolutas > 0.5):")
significant_correlations_display = significant_correlations

# Identify columns with high correlation to target variable (e.g., 'Caloric Value')
target_variable = 'Caloric Value'
correlated_features = correlation_matrix[target_variable][abs(correlation_matrix[target_variable]) > correlation_threshold]
correlated_features = correlated_features.drop(target_variable).sort_values(ascending=False)

print(f"\nCaracterísticas correlacionadas con '{target_variable}':")
correlated_features_display = correlated_features

# Recommendations for machine learning model
print("\nRecomendaciones de Columnas para Modelos de Machine Learning:")
input_features = list(correlated_features.index)
print("Características de Entrada (Input Features):", input_features)
print("Variable Objetivo (Target Variable):", target_variable)

# Display selected input features and target variable
print("\nPrimeras filas de las Características Seleccionadas y la Variable Objetivo:")
selected_features_display = data[input_features + [target_variable]].head()

data_info, first_rows, data_description, correlation_matrix_display, significant_correlations_display, correlated_features_display, input_features, target_variable, selected_features_display

ModuleNotFoundError: No module named 'seaborn'

## Respuestas a las Preguntas

**Revisar detalladamente el dataset y sacar conclusiones del mismo**

**Tipos de datos**:
- El dataset contiene columnas con tipos de datos enteros (`int64`), flotantes (`float64`) y cadenas de texto (`object`).
- Los datos de tipo `int` son solo inidices e numeradores
- Los datos de tipo flotante son datos de mediciones de micro-nutrientes

**Cantidad de columnas (campos)**:
- El dataset tiene un total de 37 columnas.

**Cantidad de filas (registros)**:
- El dataset tiene un total de 551 filas.

**Estadísticas del dataset (media, moda, desviación estándar, máximo, mínimo, etc)**:




In [4]:
data_description

Unnamed: 0.1  Unnamed: 0  Caloric Value         Fat  Saturated Fats  \
count    551.000000  551.000000     551.000000  551.000000      551.000000   
mean     275.000000  275.000000     237.359347   10.766933        3.722715   
std      159.204271  159.204271     199.235598   12.544196        5.397021   
min        0.000000    0.000000       3.000000    0.000000        0.000000   
25%      137.500000  137.500000      94.500000    2.500000        0.500000   
50%      275.000000  275.000000     186.000000    6.600000        1.800000   
75%      412.500000  412.500000     337.000000   14.400000        4.800000   
max      550.000000  550.000000    1578.000000   87.500000       43.500000   

       Monounsaturated Fats  Polyunsaturated Fats  Carbohydrates      Sugars  \
count            551.000000            551.000000     551.000000  551.000000   
mean               4.001564              2.170793      15.819036    2.708907   
std                5.540608              3.470820      20.289320    7.455666   
min                0.000000              0.000000       0.000000    0.000000   
25%                0.600000              0.400000       0.000000    0.000000   
50%                2.200000              1.100000       5.100000    0.000000   
75%                5.150000              2.600000      28.700000    2.200000   
max               48.000000             40.100000     128.300000   70.800000   

          Protein  ...      Calcium      Copper        Iron   Magnesium  \
count  551.000000  ...   551.000000  551.000000  551.000000  551.000000   
mean    18.417857  ...    94.848290    8.579102    1.530819   34.549289   
std     18.932404  ...   170.568662   44.379145    1.883603   45.296399   
min      0.000000  ...     0.000000    0.000000    0.000000    0.000000   
25%      5.300000  ...     9.650000    0.049000    0.300000    6.500000   
50%     11.900000  ...    38.300000    0.100000    1.000000   21.100000   
75%     25.650000  ...   101.400000    0.300000    2.200000   47.100000   
max     86.900000  ...  1283.500000  668.600000   21.100000  376.200000   

        Manganese   Phosphorus    Potassium     Selenium        Zinc  \
count  551.000000   551.000000   551.000000   551.000000  551.000000   
mean     3.537002   219.735563   347.234483    35.009212    1.652964   
std     12.595878   243.550965   380.140492   153.034744    6.519636   
min      0.000000     0.000000     0.000000     0.000000    0.000000   
25%      0.048000    50.250000    90.000000     0.021000    0.300000   
50%      0.100000   134.100000   223.200000     0.050000    0.800000   
75%      0.500000   317.450000   456.100000     0.084000    1.700000   
max    111.600000  1385.100000  3198.400000  1679.000000  147.300000   

       Nutrition Density  
count         551.000000  
mean          144.663207  
std           187.047623  
min             1.025000  
25%            39.933500  
50%            84.100000  
75%           170.691500  
max          1337.000000  

[8 rows x 36 columns]


**Breve descripción de lo que representan cinco columnas del dataset**:
- `Caloric Value`: Representa la cantidad de calorías en el alimento, medida en kilocalorías (kcal).
- `Fat`: Indica la cantidad total de grasa en el alimento, medida en gramos (g).
- `Carbohydrates`: Mide la cantidad de carbohidratos presentes en el alimento, en gramos (g).
- `Protein`: Representa la cantidad de proteínas en el alimento, en gramos (g).
- `Sodium`: Indica la cantidad de sodio en el alimento, medida en miligramos (mg).

**Pensar en qué columnas se podrían usar como entrada para un modelo de machine learning que resuelva un problema en el contexto del dataset (Recomendar una tarjeta de crédito, predecir ingresos, detectar desnutrición, etc)**:
- Para un modelo de machine learning que tenga como objetivo predecir el valor nutricional de un alimento o recomendar alimentos saludables, las siguientes columnas podrían ser usadas como características de entrada:
  - `Fat`
  - `Carbohydrates`
  - `Protein`
  - `Sodium`
  - `Cholesterol`
- La variable objetivo en este caso sería `Caloric Value`.

**Características de Entrada (Input Features)**:
- `Fat`
- `Carbohydrates`
- `Protein`
- `Sodium`
- `Cholesterol`

**Variable Objetivo (Target Variable)**:
- `Caloric Value`
